pkg/platform/model/buildplanner.go

```diff
@@ -345,3 +349,17 @@ func (bp *BuildPlanner) GetBuildExpression(owner, project, commitID string) (*bu
 
 	return expression, nil
 }
+
+// processBuildPlannerError will check for special error types that should be
+// handled differently. If no special error type is found, the fallback message
+// will be used.
+func processBuildPlannerError(bpErr error, fallbackMessage string) error {
+	graphqlErr := &graphql.GraphErr{}
+	if errors.As(bpErr, graphqlErr) {
+		code, ok := graphqlErr.Extensions[codeExtensionKey].(string)
+		if ok && code == clientDeprecationErrorKey {
+			return locale.NewError("err_buildplanner_deprecated", "Encountered deprecation error: {{.V0}}, please update the State Tool and try again. If the problem persists, please contact support.", graphqlErr.Message)
```

Comment:
```suggestion
			return locale.NewError("err_buildplanner_deprecated", "Encountered deprecation error: {{.V0}}", graphqlErr.Message)
```

I'd rather we leave the remediation to the deprecation message itself.